internal/pkg/client/cache/dir.go

```diff
@@ -22,75 +46,268 @@ const (
 	// for image downloads to be cached in
 	DirEnv = "SINGULARITY_CACHEDIR"
 
-	// RootDefault specifies the directory inside of ${HOME} that images are
-	// cached in by default.
-	// Uses "~/.singularity/cache" which will not clash with any 2.x cache
+	// BasedirDefault specifies the default value of the cache base directory. The
+	// path of the user's home directory is prepended at runtime. Ultimately, the
+	// default cache base directory is ~/.singularity/cache".
+	// Using "~/.singularity/cache" also does not clash with any 2.x cache
 	// directory.
-	RootDefault = ".singularity/cache"
+	BasedirDefault = ".singularity"
+
+	// RootDefault is the default cache root directory created within the base
+	// directory. This value is not supposed to be set by the user.
+	rootDefault = "cache"
+
+	// StateInitialized represents the state of a give cache after successful
+	// initialization
+	StateInitialized = "initialized"
+
+	// StateInvalid represents the state of an invalid cache
+	StateInvalid = "invalid"
```

Comment:
this value is only used in tests, it's never used in the code for this package. To me that means that the zero value of the State field should signify invalid, and this is not needed. Since this is exported and the state field is exported, this also means that the client code could set State to StateInvalid, which is probably not useful.

What this means is that State should be a boolean since there are only two states with opposite meanings.

But please see below first.

Reply:
Sorry, I had never noticed that these comments are not presented in file-order but in write-order. I wrote this after the comment on line 87. "below" refers to that.

Reply:
Yes one of the main goals is to ease testing, that is 100% correct. I will switch to a boolean, the semantic remains the same.

Reply:
The code is now using a boolean for `State`